environment/src/main/java/com/sequenceiq/environment/environment/repository/EnvironmentRepository.java

```diff
@@ -55,4 +55,8 @@
 
     @CheckPermission(action = ResourceAction.READ)
     List<Environment> findAllByStatusInAndArchivedIsFalse(Collection<EnvironmentStatus> statuses);
+
+    @CheckPermission(action = ResourceAction.READ)
+    @Query("SELECT e.resourceCrn FROM Environment e WHERE e.name = :name AND e.accountId = :accountId")
```

Comment:
is this query is necessary?

Reply:
yes, spring data jpa query methods do not support projection